utils/tools.go

```diff
@@ -200,3 +200,29 @@ func ConvertIntToByte32(n int) ([32]byte, error) {
 func ConvertByte32ToInt(nb [32]byte) int {
 	return int(binary.BigEndian.Uint64(nb[:]))
 }
+
+// ConvertProofForEthereum converts a proof to 32 byte format needed by ethereum
+func ConvertProofForEthereum(sortedHashes [][]byte) ([][32]byte, error) {
+	var property [][32]byte
+	for _, hash := range sortedHashes {
+		hash32, err := SliceToByte32(hash)
+		if err != nil {
+			return nil, err
+		}
+		property = append(property, hash32)
+	}
+
+	return property, nil
+}
+
+// GenerateHashPair generates a preimage and hash pair
```

Comment:
maybe we should add some extra comment here, explaining why are we generating the next version based on the hash of pre-image.

In the literature, it is a bit related to a concept called commit-reveal scheme.

Reply:
I wouldn't explain version stuff here as this function is general, but I have added a brief explanation on coredoc